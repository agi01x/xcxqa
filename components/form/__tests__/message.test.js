import React from 'react';
import { mount } from 'enzyme';
import Form from '..';

describe('Form', () => {
  it('should display two message', () => {
    const rules = [{
      pattern: /^\w+$/,
      message: 'Error message 1',
    }, {
      pattern: /^\w+$/,
      message: 'Error message 2',
    }];
    let myForm;
    const Form1 = Form.create()(({ form }) => {
      myForm = form;
      return (
        <Form>
          <Form.Item label="Account">
            {form.getFieldDecorator('account', { initialValue: '+=-/', rules })(<input />)}
          </Form.Item>
        </Form>
      );
    });

    const wrapper = mount(<Form1 />);
    myForm.validateFields();

    wrapper.update();
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('should display custom message', () => {
    const rules = [{
      pattern: /^$/,
      message: (<span>Account does not exist, <a rel="noopener noreferrer" href="https://www.alipay.com/" target="_blank">Forgot account?</a></span>),
    }];
    let myForm;
    const Form1 = Form.create()(({ form }) => {
      myForm = form;
      return (
        <Form>
          <Form.Item label="Account">
            {form.getFieldDecorator('account', { initialValue: 'antd', rules })(<input />)}
          </Form.Item>
        </Form>
      );
    });

    const wrapper = mount(<Form1 />);
    myForm.validateFields();

    wrapper.update();
    expect(wrapper.render()).toMatchSnapshot();
  });
});
