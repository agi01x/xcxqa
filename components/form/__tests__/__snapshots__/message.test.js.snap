// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should display custom message 1`] = `
<form
  class="ant-form ant-form-horizontal"
>
  <div
    class="ant-row ant-form-item ant-form-item-with-help"
  >
    <div
      class="ant-form-item-label"
    >
      <label
        class=""
        for="account"
        title="Account"
      >
        Account
      </label>
    </div>
    <div
      class="ant-form-item-control-wrapper"
    >
      <div
        class="ant-form-item-control has-error"
      >
        <span
          class="ant-form-item-children"
        >
          <input
            data-__field="[object Object]"
            data-__meta="[object Object]"
            id="account"
            value="antd"
          />
        </span>
        <div
          class="ant-form-explain show-help-enter"
        >
          <span>
            Account does not exist, 
            <a
              href="https://www.alipay.com/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Forgot account?
            </a>
          </span>
        </div>
      </div>
    </div>
  </div>
</form>
`;

exports[`Form should display two message 1`] = `
<form
  class="ant-form ant-form-horizontal"
>
  <div
    class="ant-row ant-form-item ant-form-item-with-help"
  >
    <div
      class="ant-form-item-label"
    >
      <label
        class=""
        for="account"
        title="Account"
      >
        Account
      </label>
    </div>
    <div
      class="ant-form-item-control-wrapper"
    >
      <div
        class="ant-form-item-control has-error"
      >
        <span
          class="ant-form-item-children"
        >
          <input
            data-__field="[object Object]"
            data-__meta="[object Object]"
            id="account"
            value="+=-/"
          />
        </span>
        <div
          class="ant-form-explain show-help-enter"
        >
          Error message 1 Error message 2
        </div>
      </div>
    </div>
  </div>
</form>
`;
